desc/compute/utils.py

```diff
@@ -114,17 +114,29 @@ def _compute(names, params, transforms, profiles, data=None, **kwargs):
                 data=data,
                 **kwargs,
             )
-            data = data_index[name]["fun"](params, transforms, profiles, data, **kwargs)
+            if transforms["grid"].axis.size:
```

Comment:
does this work correctly?
I'm thinking of the following scenario:
- user wants some data at the axis
- above block triggers, computing the regular data dependencies, but maybe doesn't have the required data to compute the axis part yet so thats nan
- this block triggers, asking to compute the axis dependencies, but if they were already computed above they get skipped, leaving us with nans at the axis?

Reply:
I think my modification on line 372 of the function `_has_dependencies` accounts for this. The inside of this if statement is executed whenever a dependency is missing (whether that be a normal dependency or one that is only needed to evaluate the limit at the axis). The `_has_dependencies` function contains logic that states quantities that are only required to evaluate the limit are considered dependencies if and only if `transforms["grid"].axis.size > 0`.

If we always use `get_data_deps(keys, has_axis=grid.axis.size > 0)` as our input to the compute function, then the `if transforms["grid"].axis.size"` is redundant. I've kept it because I think it improves readability and prevents computing the limit dependencies when they are not needed (for example if the user used `get_data_deps(keys, has_axis=not(grid.axis.size > 0))`.